### PR TITLE
Fixed tuple TypeError for nets with axis list.

### PR DESCRIPTION
**Pull request by BlaiseRitchie**
Encountered .h5 file where the layer axis was saved as 1-length list of int rather than a simple int.
Fixed the following exception:
Traceback (most recent call last):
  File "c:\python36\lib\site-packages\mmdnn\conversion\keras\keras2_parser.py", line 104, in __init__
    from keras.applications.mobilenet import relu6
ImportError: cannot import name 'relu6'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python36\Scripts\mmconvert-script.py", line 11, in <module>
    load_entry_point('mmdnn==0.2.3', 'console_scripts', 'mmconvert')()
  File "c:\python36\lib\site-packages\mmdnn\conversion\_script\convert.py", line 102, in _main
    ret = convertToIR._convert(ir_args)
  File "c:\python36\lib\site-packages\mmdnn\conversion\_script\convertToIR.py", line 46, in _convert
    parser = Keras2Parser(model)
  File "c:\python36\lib\site-packages\mmdnn\conversion\keras\keras2_parser.py", line 120, in __init__
    'DepthwiseConv2D': layers.DepthwiseConv2D
  File "c:\python36\lib\site-packages\keras\engine\saving.py", line 419, in load_model
    model = _deserialize_model(f, custom_objects, compile)
  File "c:\python36\lib\site-packages\keras\engine\saving.py", line 225, in _deserialize_model
    model = model_from_config(model_config, custom_objects=custom_objects)
  File "c:\python36\lib\site-packages\keras\engine\saving.py", line 458, in model_from_config
    return deserialize(config, custom_objects=custom_objects)
  File "c:\python36\lib\site-packages\keras\layers\__init__.py", line 55, in deserialize
    printable_module_name='layer')
  File "c:\python36\lib\site-packages\keras\utils\generic_utils.py", line 147, in deserialize_keras_object
    list(custom_objects.items())))
  File "c:\python36\lib\site-packages\keras\engine\network.py", line 1033, in from_config
    process_node(layer, node_data)
  File "c:\python36\lib\site-packages\keras\engine\network.py", line 991, in process_node
    layer(unpack_singleton(input_tensors), **kwargs)
  File "c:\python36\lib\site-packages\keras\engine\base_layer.py", line 432, in __call__
    self.build(unpack_singleton(input_shapes))
  File "c:\python36\lib\site-packages\keras\layers\normalization.py", line 95, in build
    dim = input_shape[self.axis]
TypeError: tuple indices must be integers or slices, not list

### PR Overview

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [ ] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by fchollet** [comment]
> ImportError: cannot import name 'relu6'

This error is unrelated.

File: tests/test_model_saving.py [hunk 1]
```diff
@@ -779,6 +779,23 @@ def test_saving_overwrite_option_gcs():
         file_io_proxy.delete_file(gcs_filepath)  # cleanup
 
 
+def test_saving_loading_batchnormalization_axis():
```
[the remark this answers is not in the document]
Comment (BlaiseRitchie) [reply tests/test_model_saving.py]: Odd, it doesn't work for me before this PR; did you actually run it? The axis parameter on the BatchNormalization config gets saved as a list.
`
>>> model=Sequential([
...         BatchNormalization(axis = 1, input_shape = (4, 4, 4)),
...     ])
>>> fp, fname = tempfile.mkstemp('.h5')
>>> os.close(fp)
>>> model.save(fname)
2019-01-14 08:26:53.739737: I T:\src\github\tensorflow\tensorflow\core\platform\cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2
>>> f=h5py.File(fname, 'r')
>>> json.loads(f.attrs['model_config'])['config'][0]['config']['axis']
[1]
`

[the remark this answers is not in the document]
Comment (fchollet) [reply tests/test_model_saving.py]: Yes, this code snippet can definitely be run without this PR. What goes into the config is `self.axis` which is set to what the user passed.

Perhaps it was a model that was saved using `tf.keras` that failed in your case? `tf.keras` accepts a tuple as `axis` arg, but that's not supported here (we should support it though, if you want to take that on...)

[the remark this answers is not in the document]
Comment (BlaiseRitchie) [reply tests/test_model_saving.py]: It's definitely a model that was saved with keras-team/keras. Although it seems to only start causing problems if I create models with Conv2Ds either before/after the BatchNormalization and then run the code in my previous comment or in the PR, so like:
```
>>> fp, fname = tempfile.mkstemp('.h5')
>>> os.close(fp)
>>> del fp
>>> model = Sequential([
...     Conv2D(2, (2, 2), input_shape = (4, 4, 4)),
...     BatchNormalization(axis = 1),
...     Conv2D(2, (2, 2)),
... ])
>>> f=h5py.File(fname, 'r')
>>> json.loads(f.attrs['model_config'])['config']['layers'][1]['config']['axis']
[1]
```
But once I've done that once, I can close and reopen a new Python session and re-run the simple case with only BatchNormalization and it still saves axis as a list and fails.

[the remark this answers is not in the document]
Comment (gabrieldemarmiesse) [reply tests/test_model_saving.py]: @BlaiseRitchie, please update the unit test with a piece of code which doesn't run without this PR and I'll take a look at it.

**Comment by fchollet** [comment]
Closing outdated PR.

**Pull request closed without merging**